integration_tests/e2e/updatePrisonerContactRestriction.cy.ts

```diff
@@ -0,0 +1,139 @@
+import Page from '../pages/page'
+import TestData from '../../server/routes/testutils/testData'
+import { StubPrisonerContactRestrictionDetails } from '../mockApis/contactsApi'
+import EnterRestrictionPage from '../pages/enterRestrictionPage'
+import ManageContactDetailsPage from '../pages/manageContactDetails'
+
+context('Update Prisoner Contact Restriction', () => {
+  const contactId = 654321
+  const prisonerContactId = 987654
+  const restrictionId = 555333
+  const contact = TestData.contact({
+    id: contactId,
+    lastName: 'Last',
+    firstName: 'First',
+    middleNames: 'Middle Names',
+  })
+  const prisonerContactRestriction = TestData.getPrisonerContactRestrictionDetails({
+    prisonerContactRestrictionId: restrictionId,
+    contactId: contact.id,
+    restrictionType: 'CHILD',
+    startDate: '2024-01-01',
+    expiryDate: '2050-08-01',
+    comments: 'Keep an eye',
+  })
+  const prisoner = TestData.prisoner()
+  const { prisonerNumber } = prisoner
+  const enterPageTitle = 'Update a prisoner-contact restriction'
+
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubComponentsMeta')
+    cy.task('stubSignIn', { roles: ['PRISON'] })
+    cy.task('stubRestrictionTypeReferenceData')
+    cy.task('stubTitlesReferenceData')
+    cy.task('stubPrisonerById', prisoner)
+    cy.task('stubGetContactById', contact)
+    cy.task('stubGetPrisonerContactRelationshipById', {
+      id: prisonerContactId,
+      response: TestData.prisonerContactRelationship(),
+    })
+    cy.task('stubGetPrisonerContactRestrictions', {
+      prisonerContactId,
+      response: { prisonerContactRestrictions: [prisonerContactRestriction] },
+    })
+    cy.signIn()
+
+    // TODO visit here from the prisoner contact page instead of directly
+    cy.visit(
+      `/prisoner/${prisonerNumber}/contacts/${contactId}/relationship/${prisonerContactId}/restriction/update/PRISONER_CONTACT/enter-restriction/${restrictionId}?returnUrl=/prisoner/${prisonerNumber}/contacts/manage/${contactId}/relationship/${prisonerContactId}`,
+    )
```

Comment:
Is this the part that is waiting for Prabash's PR?

Reply:
Yeah I also had to implement the GET for prisoner-contact restrictions which I know he'll be doing as well but that let me test and check it in, shouldn't be too bad. 
It's also supposed to return to the restriction tab after an update so that'll need figuring out.